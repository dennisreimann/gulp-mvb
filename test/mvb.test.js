/* globals expect, test */
const { join } = require('path')
const MVB = require('../mvb')

test('#loadArticles', () => {
  const options = {
    glob: join(__dirname, 'fixtures', '*.md'),
    permalink: article => `/articles/${article.id}.html`
  }
  const mvb = MVB(options)

  expect(mvb.articles.length).toEqual(3)

  expect(mvb.articles[0].title).toEqual('Hello World')
  expect(mvb.articles[0].subtitle).toEqual('Starting my new blog')
  expect(mvb.articles[0].description).toEqual(
    'Hello everyone, Iâ€™m starting a new blog!'
  )
  expect(mvb.articles[0].date).toEqual(new Date('2017-10-13'))
  expect(mvb.articles[0].content).toMatchSnapshot()

  expect(mvb.articles[1].title).toEqual('My second article')
  expect(mvb.articles[1].description).toEqual('This is going to be a thing!')
  expect(mvb.articles[1].date).toEqual(new Date('2017-10-14 12:34:56.000Z'))
  expect(mvb.articles[1].content).toMatchSnapshot()

  expect(mvb.articles[2].title).toEqual('No date')
  expect(mvb.articles[2].description).toBe(undefined)
  expect(mvb.articles[2].date).toBe(undefined)
  expect(mvb.articles[2].content).toMatchSnapshot()
})
