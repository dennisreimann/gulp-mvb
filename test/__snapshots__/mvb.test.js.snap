// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#loadArticles 1`] = `
"<p>Hello everyone, I’m starting a new blog!</p>
<div id="more"></div>
<p>This is the first article.</p>
"
`;

exports[`#loadArticles 2`] = `
"<p>Wow, I am totally excited – day two, second article already!</p>
"
`;

exports[`#loadArticles 3`] = `
"<p>It’s as simple as that.</p>
"
`;
